src/node_native_module.h

```diff
@@ -48,17 +62,28 @@ class NativeModuleLoader {
   void LoadCodeCacheHash();  // Loads data into code_cache_hash_
 
   v8::ScriptCompiler::CachedData* GetCachedData(const char* id) const;
-  static v8::Local<v8::Value> CompileAsModule(Environment* env,
-                                              const char* id,
-                                              bool produce_code_cache);
-  // TODO(joyeecheung): make this public and reuse it to compile bootstrappers.
+
+  // Compile a script as a NativeModule that can be loaded via
+  // NativeModule.p.require in JS land. If returns_code_cache is true,
+  // returns the cache data in a Uint8Array, otherwise returns the compiled
+  // function.
+  // TODO(joyeecheung): it's possible to always produce code cache
+  // on the main thread and consume them in worker threads, or just
+  // share the cache among all the threads, although
+  // We need to decide whether to do that even when workers are not used.
+  static v8::MaybeLocal<v8::Value> CompileAsModule(Environment* env,
+                                                   const char* id,
+                                                   bool return_code_cache);
```

Comment:
I realize you didn't introduce it but the bool would be clearer at call sites as an enum.

Reply:
Good idea (In fact, it was me who introduced this bool, just not in this patch)